rllib/env/multi_agent_env_runner.py

```diff
@@ -414,17 +414,16 @@ def _sample_episodes(
 
         done_episodes_to_return: List[MultiAgentEpisode] = []
 
-        # Reset the environment.
-        # TODO (simon): Check, if we need here the seed from the config.
-        obs, infos = self.env.reset()
-
         # Create a new multi-agent episode.
         _episode = self._new_episode()
         self._make_on_episode_callback("on_episode_created", _episode)
         _shared_data = {
             "agent_to_module_mapping_fn": self.config.policy_mapping_fn,
         }
 
+        # Reset the environment.
```

Comment:
Hey @simonsays1980 , thanks for looking into this problem. I think this issue in general here is unfortunately more complex that what meets the eye right now :( . Let me explain:
* On single-agent, users are currently not even allowed to override the on_episode_created callback :D . This is because in single-agent, we use gym's vector env, which resets envs automatically after a terminal is hit, which makes it impossible to call the on_episode_created callback before this auto-reset happens. See [here](https://github.com/ray-project/ray/blob/master/rllib/algorithms/algorithm_config.py#L4201) and [here](https://github.com/ray-project/ray/blob/master/rllib/env/single_agent_env_runner.py#L361).
* For multi-agent (where currently we don't use gym.vector) this does actually work and I therefore would suggest, we only fix this for now on the multi-agent env runner.
* In MultiAgentEnvRunner, however, we should then also fix it for `_sample_episodes()`.
* We should update the docstring in callbacks.py to reflect that this callback is NOT currently valid for new API stack + single-agent.
* We should remove the on_episode_created callback call entirely from single-agent env runner.